lib/reporting/report_headers_builder.rb

```diff
@@ -18,12 +18,12 @@ def available_headers
       report.columns.keys.map { |key| [translate_header(key), key] }
     end
 
-    def fields_to_hide
+    def fields_to_show
```

Comment:
Hi @jibees, I think what you are missing is this method. It should be changed to something like

```
fields_in_headers = if report.display_header_row?
                      report.formatted_rules.map { |rule| rule[:fields_used_in_header] }.flatten.reject(&:blank?)
                    else
                      []
                    end
params_fields_to_show - fields_in_headers
```